JustSaying/Fluent/DefaultServiceResolver.cs

```diff
@@ -0,0 +1,89 @@
+using System;
+using JustSaying.AwsTools;
+using JustSaying.Messaging.MessageSerialization;
+using JustSaying.Messaging.Monitoring;
+using Microsoft.Extensions.Logging;
+
+namespace JustSaying.Fluent
+{
+    /// <summary>
+    /// A class representing the built-in implementation of <see cref="IServiceResolver"/>. This class cannot be inherited.
+    /// </summary>
+    internal sealed class DefaultServiceResolver : IServiceResolver
+    {
+        /// <inheritdoc />
+        public T ResolveService<T>()
+            => (T)ResolveService(typeof(T));
+
+        private object ResolveService(Type desiredType)
+        {
+            if (desiredType == typeof(ILoggerFactory))
+            {
+                return new NullLoggerFactory();
+            }
+            else if (desiredType == typeof(IAwsClientFactoryProxy))
+            {
+                return new AwsClientFactoryProxy();
+            }
+            else if (desiredType == typeof(IHandlerResolver))
+            {
+                return null; // Special case - must be provided by the consumer
+            }
+            else if (desiredType == typeof(IMessagingConfig))
+            {
+                return new MessagingConfig();
+            }
+            else if (desiredType == typeof(IMessageMonitor))
+            {
+                return new NullOpMessageMonitor();
+            }
+            else if (desiredType == typeof(IMessageSerializationFactory))
+            {
+                return new NewtonsoftSerializationFactory();
+            }
+            else if (desiredType == typeof(IMessageSerializationRegister))
+            {
+                return new MessageSerializationRegister(ResolveService<IMessageSubjectProvider>());
+            }
+            else if (desiredType == typeof(IMessageSubjectProvider))
+            {
+                return new NonGenericMessageSubjectProvider();
```

Comment:
Seeing that we are changing a whole load of things, I think we should return GenericMessageSubjectProvider by default

Reply:
My understanding is that this is as-is because it needs to be backwards compatible for the wire format/queue naming, rather than the C# code.

Reply:
Possibly one to discuss in a separate PR, but it looks like `GenericMessageSubjectProvider` would be a better default, and compatible with the default, it only extends and fixes the cases where the type name exceeds 100 chars, or the type is generic